lib/three_scale/middleware/multitenant.rb

```diff
@@ -111,9 +111,27 @@ def call(env)
       def _call(env)
         Thread.current[:multitenant] = TenantChecker.new(@attribute,@app,env)
         @app.call(env)
+      rescue ActionDispatch::ParamsParser::ParseError => error
```

Comment:
I do not think it is the responsibility of this middleware. Rather create another one that is called just before ActionDis

It should be caught earlier in the chain, create another middleware if needed, so you will disable it on test and development but safely rescue in production

Reply:
> so you will disable it on test and development but safely rescue in production
I do not think that disabling it by default in `test` is a good idea 😄 We should make valid format requests in `test`, OR test that it is behaving as we desire in production when the params are not correct.

>I do not think it is the responsibility of this middleware. Rather create another one that is called just before Rack::Sendfile
> It should be caught earlier in the chain, create another middleware if needed

Gotcha 👍

Reply:
Raising in development and test is better no? In production you fallback to a response.

That's how rails work with 404 and `ActiveRecord::RecordNotFound` at least. Make it configurable with sane defaults

Reply:
For everybody else, we spoke about this in private, and these are the conclusions:

We never raise it again for RecordNotFound and we won't raise it either for this one 😄 

https://github.com/3scale/porta/blob/7b55693bcedfed8cd707550126c1444a0187111d/app/controllers/application_controller.rb#L11
https://github.com/3scale/porta/blob/7b55693bcedfed8cd707550126c1444a0187111d/app/controllers/concerns/error_handling.rb#L8
https://github.com/3scale/porta/blob/7b55693bcedfed8cd707550126c1444a0187111d/app/controllers/concerns/error_handling.rb#L22-L24
https://github.com/3scale/porta/blob/7b55693bcedfed8cd707550126c1444a0187111d/app/controllers/concerns/error_handling.rb#L40-L65